### ISSUE TITLE
how to install the CEP extension properly

### ISSUE THREAD

**Issue by xuanywn** [Adobe-CEP/Getting-Started-guides#47]
## Topic

This is an issue regarding:

How to install the cep extension properly, without enabling debugger mode by modifying the registry.
How to understand the description in the document：
 If you aren't distributing on Adobe Exchange or the Admin Console, ZXP may not be required

## Versions

- [ ] CEP version(s) used 11（ cep 6）
- [ ] Supported CC app(s) and version(s):   premiere pro 24.1.0

## Description of the issue
question 1：

I want to know how to package and install a cep extension, It is not necessary to package it as a ZXP according to the decscription in the document. If not packaged as a ZXP file, how can it be installed properly?
![image](https://github.com/user-attachments/assets/20460ba7-8e87-488f-b2d8-252e3f74fe48)

question 2：
After signing and packaging with a self-signed certificate, I got a ZXP package , baichuanPlugin.zxp, but it cannot be installed.
I execute the following command , but there was no output:
```bash

D:\permiere\ExManCmd_win>ExManCmd --install  D:\permiere\CEP-Resources-master\ZXPSignCMD\4.1.1\win64\baichuanPlugin.zxp

D:\permiere\ExManCmd_win>
```

further, I try to install a cep extension by Anastasiy’s extension manager, and it gave a tip like below:
![image](https://github.com/user-attachments/assets/56758d3f-ec2c-4c0b-b6fc-0d2ea8b09983)
The attachment contains the code for the plugin.
[PProPanel.zip](https://github.com/user-attachments/files/16973145/PProPanel.zip)

In conclusion, I want to know how to install the CEP extension properly.

I would appreciate some help, and would be grateful if someone could respond!

**Comment by xuanywn** [comment]
help wanted

**Comment by ErinFinnegan** [comment]
Hi @xuanywn, this isn't the best place to ask for support.

**_Summary:_** You can essentially dump the files inside the ZXP (which is actually a ZIP) in a valid location and it may work, at least until a product update, which might delete the files because it wasn't installed in the expected method.

Anastasiy's error message would indicate that you don't seem to have the expected compatible host app installed. That means that the manifest may not have the correct product versions listed. Please [check this chart](https://github.com/Adobe-CEP/CEP-Resources/blob/master/CEP_11.x/Documentation/CEP%2011.1%20HTML%20Extension%20Cookbook.md#applications-integrated-with-cep). For newer products, you may be able to see[ the correct versions here](https://github.com/Adobe-CEP/CEP-Resources/blob/CEP-12-Prerelease/CEP_12.x/Documentation/CEP%2012%20HTML%20Extension%20Cookbook.md) (however, this doesn't seem to be updated for 2024 yet).

That said, I started to write a guide to installing [the CEP test extension](https://github.com/Adobe-CEP/CEP-Resources/tree/master/CEP_11.x/Samples/CEP_HTML_Test_Extension-10.0) since the installation documentation is a bit scarce. Downloading and installing the CEP Test extension is a good "sanity check", because if you can get it running you can verify if something is wrong with your ZXP.

Perhaps this will help:

## Step One: Download the Test Extension

### Download Methods

If you are new to github, how to dowload the sample itself may not be readily apparent. There are two ways to do this:

#### Download Method One
Download the entire `CEP-Resources` repo and navigate to the sample folder you wish to install. At the time of this writing the repo is over 16 GB, and this may be undesirable on slow connections.

#### Download Method Two
You may download the sample files individually and recreate the test extension's directory structure on your machine.

## Step Two: Move the directory into a valid CEP location

After successfully downloading the `CEP_HTML_Test_Extension`, you will need to move it into a valid CEP location on your system. The full documentation on these locations can be found [in the CEP Cookbook](https://github.com/Adobe-CEP/CEP-Resources/blob/master/CEP_11.x/Documentation/CEP%2011.1%20HTML%20Extension%20Cookbook.md#extension-folders).

Use one of the two locations below. Note that the System extension folder may be invisible by default, and you may need administrator privaledges to install there:

- System extension folder
    - Win(x64): `C:\Program Files (x86)\Common Files\Adobe\CEP\extensions`, and `C:\Program Files\Common Files\Adobe\CEP\extensions`
    - macOS: `/Library/Application Support/Adobe/CEP/extensions`

- Per-user extension folder
    - Win: `C:\Users\<USERNAME>\AppData\Roaming\Adobe\CEP/extensions`
    - macOS: `~/Library/Application Support/Adobe/CEP/extensions`

## Step Three: Launch the Creative Cloud app you wish to test with

If you had the application you wish to test with open during step two above, you will need to close and reopen the application.

The applications supported by the HTML test extension are listed in the `manifest.xml` file, located in the `/CSXS` directory:

```
<HostList>
    <Host Name="DRWV" Version="15.0"/>
    <Host Name="FLPR" Version="15.0"/>
    <Host Name="IDSN" Version="11.0"/>
    <Host Name="AICY" Version="11.0"/>
    <Host Name="ILST" Version="19.0"/>
    <Host Name="PHSP" Version="16.0"/>
    <Host Name="PHXS" Version="16.0"/>
    <Host Name="PPRO" Version="9.0"/>
    <Host Name="PRLD" Version="4.0"/>
    <Host Name="AEFT" Version="13.0"/>
    <Host Name="DEMO" Version="1.0"/>
    <Host Name="AUDT" Version="8.0"/>
    <Host Name="LTRM" Version="7.0"/>
    <Host Name="MUSE" Version="2015"/>
</HostList>
```

The table of the product codes above are listed [in the CEP Cookbook](https://github.com/Adobe-CEP/CEP-Resources/blob/master/CEP_11.x/Documentation/CEP%2011.1%20HTML%20Extension%20Cookbook.md#applications-integrated-with-cep).

We can tell from the `manifest.xml` code above that this sample extension is compatible with "ILST 19" which is Illustrator 19.0 and higher, also known as Illustrator 2015. [This table displays older version numbers](https://github.com/Adobe-CEP/CEP-Resources/blob/master/CEP_9.x/Documentation/CEP%209.0%20HTML%20Extension%20Cookbook.md#applications-integrated-with-cep).

Generally speaking, Creative Cloud products will show a year when they launch, such as "Illustrator 2022". You can find the precise version number of the application you have installed by going to the "About" menu on macOS, or the "Help > About" menus on Windows.

Another way to navigate product version numbers is by using the Creative Cloud desktop app, which, at the time of this writing, displays product version numbers.

You may also read [this HelpX guide to finding Creative Cloud app versions](https://helpx.adobe.com/download-install/using/how-to-check-a-creative-cloud-app-version-.html).


## Step Four: Locate the test extension in the Creative Cloud host application

The CEP test extenstion is built to work in many different Creative Cloud apps.

Where to find the HTML Test extension may vary slightly in different applications:

- Illustrator: Window > Extensions > HTML Test Panel
- InDesign: Window > Extensions > HTML Test Panel
    * Note: You need to have a project open to see the installed extensions.
- Premiere Pro: Window > Extensions > HTML Test Panel
    * Note: You need to have a project open to see the installed extensions.
- Bridge:  Window > Extensions > Hello Bridge!
- Photoshop_*_ (Intel/Rosetta 2 mode only):  Window > Extensions (legacy) > HTML Test Panel

### Photoshop notes

_*_ Photoshop does not support CEP extensions running on Apple Silicon/M1 native. Users may still use CEP panels by [launching Photoshop in Intel mode](https://helpx.adobe.com/creative-cloud/kb/troubleshoot-common-addon-installation-issues.html#PromptedtoopenappinIntelemulationmodeafterinstallingplugin).

Photoshop extension developers are encouraged to begin migrating CEP extensions [to UXP](https://developer.adobe.com/photoshop/uxp). See our [migration guide for details](https://github.com/Adobe-CEP/CEP-Resources/tree/master/UXP-Migration-Guide).

## More information

- [Debugging](https://github.com/Adobe-CEP/Getting-Started-guides/tree/master/Client-side%20Debugging)
- [Fix common installation issues with Creative Cloud app extensions or plugins](https://helpx.adobe.com/creative-cloud/kb/troubleshoot-common-addon-installation-issues.html)
- [Working from the command line](https://helpx.adobe.com/creative-cloud/help/working-from-the-command-line.html)

**Comment by xuanywn** [comment]
[PProPanel.zip](https://github.com/user-attachments/files/17001728/PProPanel.zip)
Thank you very much for your reply.

Based on your suggestions, I packaged the CEP test extension as a ZXP file, but I still could not install it.

My main issue now is that I cannot install the ZXP package through the command line or Anastasiy’s Extension Manager.

Under the debug mode (by adding "PlayerDebugMode" key to the registry), the test package work well. However, after packaging it into a ZXP file, the installation is not successful.
It still shows the tip:
![image](https://github.com/user-attachments/assets/febaf1c2-9ec1-4599-89ec-f14455492bef)

Besides, I change the version to latest version in the mainfest.xml file according to [the correct versions here](https://github.com/Adobe-CEP/CEP-Resources/blob/CEP-12-Prerelease/CEP_12.x/Documentation/CEP%2012%20HTML%20Extension%20Cookbook.md) , and the installation still failed.


I am currently unsure where the issue lies, and I would appreciate your help in analyzing this.

Additionally, you mentioned: "this isn't the best place to ask for support." I would like to know the best way to seek help.

Thanks for your help again , and I'm always looking forward to your reply!!!

**Comment by ErinFinnegan** [comment]
Did you manage to successfully [sign your ZXP?](https://github.com/Adobe-CEP/Getting-Started-guides/tree/master/Package%20Distribute%20Install#use-zxpsigncmd-to-package-your-extension)? Only self-signing seems to work these days. It's also tricky to find [a working timestamp provider](https://gist.github.com/Manouchehri/fd754e402d98430243455713efada710). 
 
> Additionally, you mentioned: "this isn't the best place to ask for support." I would like to know the best way to seek help.

There isn't one, right now. CEP issues are thrown to "community support". You could ask in these forums: [https://community.adobe.com/t5/premiere-pro/ct-p/ct-premiere-pro?page=1&sort=latest_replies&filter=all&lang=all&tabid=discussions&topics=label-sdk](https://community.adobe.com/t5/premiere-pro/ct-p/ct-premiere-pro?page=1&sort=latest_replies&filter=all&lang=all&tabid=discussions&topics=label-sdk )

Failing that, you could reach out to wwds@adobe.com ... your ZIP would be caught I the spam filter, so perhaps use a link.

**Comment by xuanywn** [comment]
According to your suggestions, I tried to solve the installation issue, and it still failed, but the error information has changed

I posted the problem on the adobe community:
https://community.adobe.com/t5/premiere-pro-discussions/ppro-panel-install-error-410/m-p/14868696/thread-id/529660

and the certificate and ZXP file is :
[dayang.zip](https://github.com/user-attachments/files/17053143/dayang.zip)

The CEP extension file is:
[CEP_HTML_Test_Extension-10.0.zip](https://github.com/user-attachments/files/17053148/CEP_HTML_Test_Extension-10.0.zip)

**Comment by ErinFinnegan** [comment]
> it still failed, but the error information has changed

What is the new error?

**Comment by ErinFinnegan** [comment]
For a lack of other options, you might want to reach out to: wwds@adobe.com 

Were you able to sign and install the CEP Test Extension successfully?

**Comment by xuanywn** [comment]
I'm trying to package the test cep extension, and adjust the command-line parameters, the error imformation has changed during the installation process.

This is my operation steps below:

step 1:

generate the self-signed certificate:

```bash
D:\permiere\signpackage>ZXPSignCmd -selfSignedCert ZH BEIJING xuguohui dayang dayang123 dayang.p12
Self-signed certificate generated successfully
```

It is able to the certificate successfully.  the certificate name is dayang.p12





step 2:

Based on the previously generated certificate, I packaged the test CEP extension to a zxp file.

```bash
D:\permiere\signpackage>ZXPSignCmd -sign D:\permiere\CEP-Resources-master\CEP_11.x\Samples  dayang.zxp dayang.p12 dayang123  -tsa https://rfc3161.ai.moda
Signed successfully
```



step 3:

And I installed the zxp plugin that generated in the previously step by Anastasiy’s extension manager.

It shows the error：

![image-20240919102301939](https://github.com/user-attachments/assets/59a72256-ffb9-4952-b122-8888c02f1620)


How can I solve the problem?

the premiere pro vsion is :

![image-20240919105847915](https://github.com/user-attachments/assets/31102017-6cde-488c-ab07-04c175aa139a)

Is there need to downgrade the version of premiere pro?

**Comment by xuanywn** [comment]
> For a lack of other options, you might want to reach out to: [wwds@adobe.com](mailto:wwds@adobe.com)
> 
> Were you able to sign and install the CEP Test Extension successfully?

The CET Test Extension was successfully signed, but the installation failed.

**Comment by ErinFinnegan** [comment]
Did the Test Extension installation fail with the same error (-410)?

**Comment by ErinFinnegan** [comment]
-410 is “invalid ZXP” so something has gone wrong: https://community.adobe.com/t5/exchange-discussions/every-exman-command-error-code-exmancmd-error-code-list/td-p/11386103

**Comment by xuanywn** [comment]
Yes. I just packed and installed it with test cep extension.
---- Replied Message ----
| From | Erin ***@***.***> |
| Date | 09/21/2024 07:10 |
| To | Adobe-CEP/Getting-Started-guides ***@***.***> |
| Cc | xuanywn ***@***.***>,
Mention ***@***.***> |
| Subject | Re: [Adobe-CEP/Getting-Started-guides] how to install the CEP extension properly (Issue #47) |

Did the Test Extension installation fail with the same error (-401)?

**Comment by xuanywn** [comment]
> -410 is “invalid ZXP” so something has gone wrong: https://community.adobe.com/t5/exchange-discussions/every-exman-command-error-code-exmancmd-error-code-list/td-p/11386103


The package I created is the official Test CEP Extension code provided by you, and I followed the steps previously described, which are consistent with the methods outlined in the official documentation. The relevant test code and ZXP file were provided in my earlier response. Could you please help analyze where the issue might be? I would greatly appreciate it.
